### PR TITLE
Error handling for invalid prufer sequence `from_prufer_sequence`: issue #6420

### LINKED ISSUE #6420: rom_prufer_sequence report "ValueError: too many values to unpack"
### Current Behavior

If you try the following code
```
import networkx as nx
nx.from_prufer_sequence([7, 5, 5, 3, 1])
```
you will get
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/xing/miniconda3/lib/python3.9/site-packages/networkx/algorithms/tree/coding.py", line 396, in from_prufer_sequence
    u, v = orphans
ValueError: too many values to unpack (expected 2)
```

### Expected Behavior

The code should return a tree without any problem. See Applied Combinatorics by Mitchel T. Keller, William T. Trotter, [Chapter 5.6](https://www.rellek.net/book/s_graphs_counting-trees.html)

### PR DESCRIPTION
**Pull request by mohamedrezk122**
This pull request is meant for fixing #6420. the author of the issue passes the prufer sequence `[7,5,5,3,1]` to the function `from_prufer_sequence` and expects a valid output (tree). The conflict happens due to the difference in indexing styles; networkx uses the zero-indexing , on the other hand , mathematicians (graph theorists ) use one as the first label/index . Adopting zero-indexing style, every element in the sequence has to obey the following constraint: 
$$0 \le sequence[i] \le length(sequence) + 1   \ \ \ \  ,  \ \ \ \ such \  that  \ \ \  0 \le i < length(sequence)$$
If not the case a `NetworkXError` is raised .
Hope this is thoughtful  

Closes #6420

### CODE REVIEW

File: networkx/algorithms/tree/coding.py [hunk 1]
```diff
@@ -384,6 +389,12 @@ def from_prufer_sequence(sequence):
     not_orphaned = set()
     index = u = next(k for k in range(n) if degree[k] == 1)
     for v in sequence:
+        # check the validity of the prufer sequence
+        if v < 0 or v > n - 1:
+            msg = """Invalid Prufer sequence: you may check that every element (sequence[i]) in the sequence
+                     does not exceed the length of the sequence plus 1 or ( 0 <= sequence[i] <= length(sequence)+1 )."""
+            raise nx.NetworkXError(msg)
```
Comment (rossbar) [inline networkx/algorithms/tree/coding.py:396]: This message is a bit wordy (and would render very wide in the terminal). Perhaps it could be shortened to something like:

```python
NetworkXError(f"Values must be between 0 and len(sequence)-1, got {v}")
```

**Review by rossbar — COMMENTED** [review]
+1 for this approach, just one comment on perhaps trying to condense the exception message a bit. Thanks @mohamedrezk122

**Review by MridulS — APPROVED** [review]
@mohamedrezk122 thanks for the PR! I have shortened the error message as suggested by @rossbar

**Review by dschult — APPROVED** [review]

File: networkx/algorithms/tree/coding.py [hunk 2]
```diff
@@ -384,6 +389,11 @@ def from_prufer_sequence(sequence):
     not_orphaned = set()
     index = u = next(k for k in range(n) if degree[k] == 1)
     for v in sequence:
+        # check the validity of the prufer sequence
+        if v < 0 or v > n - 1:
+            raise nx.NetworkXError(
+                f"Invalid Prufer sequence: Values must be between 0 and {len(sequence)-1}, got {v}"
```
Comment (dschult) [inline networkx/algorithms/tree/coding.py:395]:
```suggestion
                f"Invalid Prufer sequence: Values must be between 0 and {n-1}, got {v}"
```

**Review by dschult — COMMENTED** [review]
I think `n` and `len(sequence)` are not the same. So this error message will not be correct.
I think `n` will be better.

This should also pass `black` line restrictions for 88 chars.

**Review by rossbar — APPROVED** [review]
Thanks @mohamedrezk122 @MridulS @dschult !
